File: kirun-java/src/main/java/com/fincity/nocode/kirun/engine/runtime/expression/operators/unary/ArithmeticUnaryPlusOperator.java
package com.fincity.nocode.kirun.engine.runtime.expression.operators.unary;

import com.fincity.nocode.kirun.engine.runtime.expression.Operation;
import com.fincity.nocode.kirun.engine.util.primitive.PrimitiveUtil;
import com.google.gson.JsonElement;

public class ArithmeticUnaryPlusOperator implements UnaryOperator {

	@Override
	public JsonElement apply(JsonElement t) {
		
		this.nullCheck(t, Operation.UNARY_PLUS);

		PrimitiveUtil.findPrimitiveNumberType(t.getAsJsonPrimitive());
		return t;
	}
}

File: kirun-js/src/engine/runtime/expression/operators/unary/ArithmeticUnaryPlusOperator.ts
import { PrimitiveUtil } from '../../../../util/primitive/PrimitiveUtil';
import { Operation } from '../../Operation';
import { UnaryOperator } from './UnaryOperator';

export class ArithmeticUnaryPlusOperator extends UnaryOperator {
    public apply(t: any): any {
        this.nullCheck(t, Operation.UNARY_PLUS);

        PrimitiveUtil.findPrimitiveNumberType(t);

        return t;
    }
}
